compose/src/main/java/com/bottlerocketstudios/compose/auth/Login.kt

```diff
@@ -1,16 +1,316 @@
+        ) {
+            AppIcon()
+            TextFields(state = state)
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.End
+            ) {
+                SurfaceButton(
+                    buttonText = stringResource(id = R.string.login_forgot),
+                    onClick = state.onForgotClicked,
+                    modifier = Modifier.padding(end = Dimens.grid_5)
+                )
+            }
+            Row(
+                modifier = Modifier.padding(bottom = Dimens.grid_6)
+            ) {
+                Spacer(modifier = Modifier.weight(1f))
+                Column(
+                    modifier = Modifier.weight(1f),
+                    horizontalAlignment = Alignment.CenterHorizontally
+                ) {
+                    PrimaryButton(
+                        buttonText = stringResource(id = R.string.login_button),
+                        forceCaps = true,
+                        onClick = state.onLoginClicked,
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(top = Dimens.grid_2_5)
+                    )
+                    SurfaceButton(
+                        buttonText = stringResource(id = R.string.signup_button),
+                        forceCaps = true,
+                        onClick = state.onSignupClicked,
+                        modifier = Modifier.fillMaxWidth()
+                    )
+                    OutlinedSurfaceButton(
+                        buttonText = stringResource(id = R.string.dev_options_button),
+                        forceCaps = true,
+                        onClick = state.onDevOptionsClicked,
+                        modifier = Modifier.padding(top = Dimens.grid_4)
+                    )
+                }
+                Spacer(modifier = Modifier.weight(1f))
+            }
+        }
+    }
+}
+
+@Composable
+fun AppIcon() {
+    Image(
+        modifier = Modifier.padding(top = Dimens.grid_6),
+        painter = painterResource(id = R.drawable.ic_logo),
+        contentDescription = stringResource(id = R.string.splash_image_content_description)
+    )
+}
+
+@Composable
+fun TextFields(state: LoginScreenState) {
+    MaterialTheme(
+        colors = MaterialTheme.colors.copy(
+            primary = MaterialTheme.colors.onBackground
+        )
+    ) {
+        val localFocusManager = LocalFocusManager.current
+        LoginScreenUsernameTextField(
+            modifier = Modifier
+                .padding(
+                    start = Dimens.grid_6,
+                    end = Dimens.grid_6,
+                    top = Dimens.grid_2
+                )
+                .fillMaxWidth(),
+            valueText = state.email.value,
+            onValueChange = state.onEmailChanged,
+            onNextClicked = {
+                localFocusManager.moveFocus(FocusDirection.Down)
```

Comment:
@ianslinker-br This is an interesting difference from how xml layouts handle focus. Thanks for maintaining user expectations!